File: Android/java/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/OkHttpInspectorResponse.java
package com.didichuxing.doraemonkit.kit.network.okhttp;

import com.didichuxing.doraemonkit.kit.network.core.NetworkInterpreter;

import okhttp3.Request;
import okhttp3.Response;

public class OkHttpInspectorResponse implements NetworkInterpreter.InspectorResponse {
    private final int mRequestId;
    private final Request mRequest;
    private final Response mResponse;

    public OkHttpInspectorResponse(
            int requestId,
            Request request,
            Response response) {
        mRequestId = requestId;
        mRequest = request;
        mResponse = response;
    }

    @Override
    public int requestId() {
        return mRequestId;
    }

    @Override
    public String url() {
        return mRequest.url().toString();
    }

    @Override
    public int statusCode() {
        return mResponse.code();
    }

    @Override
    public int headerCount() {
        return mResponse.headers().size();
    }

    @Override
    public String headerName(int index) {
        return mResponse.headers().name(index);
    }

    @Override
    public String headerValue(int index) {
        return mResponse.headers().value(index);
    }

    @Override
    public String firstHeaderValue(String name) {
        return mResponse.header(name);
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/OkHttpInspectorResponse.kt
package com.didichuxing.doraemonkit.kit.network.okhttp

import com.didichuxing.doraemonkit.kit.network.okhttp.core.NetworkInterpreter.InspectorResponse
import okhttp3.Request
import okhttp3.Response

class OkHttpInspectorResponse(
        private val mRequestId: Int,
        private val mRequest: Request,
        private val mResponse: Response) : InspectorResponse {
    override fun requestId(): Int {
        return mRequestId
    }

    override fun url(): String? {
        return mRequest.url().toString()
    }

    override fun statusCode(): Int {
        return mResponse.code()
    }

    override fun headerCount(): Int {
        return mResponse.headers().size()
    }

    override fun headerName(index: Int): String {
        return mResponse.headers().name(index)
    }

    override fun headerValue(index: Int): String {
        return mResponse.headers().value(index)
    }


    override fun firstHeaderValue(name: String): String? {
        return mResponse.header(name)
    }

}